Shared/Samples/Search for web map/SearchForWebMapView.Model.swift

```diff
@@ -0,0 +1,109 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+extension SearchForWebMapView {
+    /// The view model for the sample.
+    class Model: ObservableObject {
+        /// A portal to ArcGIS Online to get the portal items from.
+        private let portal = Portal.arcGISOnline(connection: .anonymous)
+        
+        /// The query parameters for the next set of results based on the last results.
+        private var nextQueryParameters: PortalQueryParameters?
+        
+        /// The text query from the last search.
+        private var lastQuery = ""
+        
+        /// The portal items resulting from a search.
+        @Published private(set) var portalItems: [PortalItem] = []
+        
+        /// Finds the portal items that match the given query.
+        /// - Parameter query: The text query used to find the portal items.
+        func findItems(for query: String) async throws {
+            // Ensure that a new search is necessary.
+            guard query != lastQuery else { return }
+            
+            if query.isEmpty {
+                await MainActor.run {
```

Comment:
While it is great to have a clear sense to only use main thread to update UI and use arbitrary threads to run other async tasks, I prefer to simply mark the whole model as `@MainActor` instead.